File: src/Unit-1/FSharp/lesson4/Completed/Messages.fs
[<AutoOpen>]
module Messages

open Akka.Actor

type ProcessCommand = 
    | ContinueProcessing

type ErrorType =
    | Null
    | Validation

type InputResult =
    | InputSuccess of reason: string
    | InputError of reason: string * errorType: ErrorType

type TailCommand =
    | StartTail of filePath: string * reporterActor: IActorRef
    | StopTail of  filePath: string

type FileCommand =
    | FileWrite of fileName: string
    | FileError of fileName: string * reason: string
    | InitialRead of fileName: string * text: string

File: src/Unit-1/DoThis/Messages.cs
namespace WinTail
{
    public class Messages
    {
        public class ContinueProcessing
        {
        }

        public class InputSuccess
        {
            public InputSuccess(string reason)
            {
                Reason = reason;
            }

            public string Reason { get; private set; }
        }

        public class InputError
        {
            public InputError(string reason)
            {
                Reason = reason;
            }

            public string Reason { get; private set; }
        }

        public class NullInputError : InputError
        {
            public NullInputError(string reason) : base(reason) { }
        }

        public class ValidationError : InputError
        {
            public ValidationError(string reason) : base(reason) { }
        }
    }
}